compiler/ballerina-lang/src/main/java/io/ballerina/projects/JBallerinaBalaWriter.java

```diff
@@ -152,17 +152,35 @@ protected void addCompilerPlugin(ZipOutputStream balaOutputStream) throws IOExce
         }
     }
 
+    /**
+     * Mark target platform as `java11` if one of the following condition fulfils.
+     * 1) Direct dependencies of imports in the package have any `ballerina/java` dependency.
+     * 2) Package has defined any platform dependency.
+     *
+     * @param pkgResolution package resolution
+     * @return target platform
+     */
     private CompilerBackend.TargetPlatform getTargetPlatform(PackageResolution pkgResolution) {
         ResolvedPackageDependency resolvedPackageDependency = new ResolvedPackageDependency(
                 this.packageContext.project().currentPackage(), PackageDependencyScope.DEFAULT);
         Collection<ResolvedPackageDependency> resolvedPackageDependencies = pkgResolution.dependencyGraph()
                 .getDirectDependencies(resolvedPackageDependency);
+
+        // 1) Check direct dependencies of imports in the package have any `ballerina/java` dependency
         for (ResolvedPackageDependency dependency : resolvedPackageDependencies) {
-            if (dependency.packageInstance().packageOrg().value().equals("ballerina")
-                    && dependency.packageInstance().packageName().value().equals("jballerina.java")) {
+            if (dependency.packageInstance().packageOrg().value().equals("ballerina") && dependency.packageInstance()
```

Comment:
Shall we use `Names.BALLERINA_ORG` and `Names.JAVA` instead of the hardcoded strings?

Reply:
Added